12-offer-encoding.md

```diff
@@ -0,0 +1,1057 @@
+  - MUST set `payment_hash` to the SHA2 256-bit hash of the
+    `payment_preimage` that will be given in return for payment.
+  - MUST set (or not set) `send_invoice` the same as the offer.
+  - MUST set `offer_id` to the id of the offer.
+  - MUST specify exactly one signature TLV: `signature`.
+    - MUST set `sig` to the signature using `node_id` as described in [Signature Calculation](#signature-calculation).
+  - if the chain for the invoice is not solely bitcoin:
+    - MUST specify `chains` the invoice is valid for.
+  - otherwise:
+    - the bitcoin chain is implied as the first and only entry.
+  - if it has bolt11 features:
+    - MUST set `features` to the bitmap of features.
+  - if the invoice corresponds to an offer with `recurrence`:
+    - MUST set `recurrence_basetime` to the start of period #0 as calculated
+      by [Period Calculation](#offer-period-calculation).
+    - if it sets `relative_expiry`:
+      - MUST NOT set `relative_expiry` `seconds_from_creation` more than the number of seconds after `created_at` that payment for this period will be accepted.
+  - otherwise:
+    - MUST not set `recurrence_basetime`.
+  - if the expiry for accepting payment is not 7200 seconds after `created_at`:
+    - MUST set `relative_expiry` `seconds_from_creation` to the number of
+      seconds after `created_at` that payment of this invoice should not be attempted.
+  - if the `min_final_cltv_expiry` for the last HTLC in the route is not 18:
+    - MUST set `min_final_cltv_expiry`.
+  - if it accepts onchain payments:
+    - MAY specify `fallbacks`
+    - MUST specify `fallbacks` in order of most-preferred to least-preferred
+      if it has a preference.
+    - for the bitcoin chain, it MUST set each `fallback_address` with
+      `version` as a valid witness version and `address` as a valid witness
+      program
+  - if it is connected only by private channels:
+    - MUST include a `blinded_path` containing one or more paths to the node.
+  - otherwise:
+    - MAY include `blinded_path`.
+  - if it includes `blinded_path`:
+    - MUST specify `path` in order of most-preferred to least-preferred if
+      it has a preference.
+    - MUST include `blindedpay` with exactly one `payinfo` for
+      each `onionmsg_path` in `blinded_path`, in order.
+    - if it includes `blinded_capacities`:
+      - MUST include exactly one `incoming_msat` (in millisatoshis) per `path`, reflecting the expected minimum amount that can be sent through the path.
+    - SHOULD ignore any payment which does not use one of the paths.
+  - otherwise:
+    - MUST NOT include `blinded_payinfo`.
+  - MUST specify `amount`.`msat` in multiples of the minimum lightning-payable unit
+    (e.g. milli-satoshis for bitcoin) for the first `chains` entry.
+  - if responding to an `invoice_request`:
+    - if for the same `offer_id`, `payer_key` and `recurrence_counter` (if any) as a previous `invoice_request`:
+      - MAY simply reuse the previous invoice.
```

Comment:
What about all the other field from the invoice request? If `amount`, `quantity`, `payer_info`, `payer_note` or `replace_invoice` have changed, it should generate a new invoice.

Reply:
Yes, hence MAY.  I've appended to the Rationale:

```
Note that such caching is optional, and should be carefully limited when e.g. currency conversion is involved.
```